src/core/container.jl

```diff
@@ -1,8 +1,8 @@
 mutable struct Trace
   task  ::  Task
   vi    ::  VarInfo
-  spl   ::  Union{Nothing, Sampler}
-  Trace() = (res = new(); res.vi = VarInfo(); res.spl = nothing; res)
+  spl   ::  AbstractSampler
```

Comment:
Maybe we can make `Trace` a parametric type?

Reply:
Done.